types/header.go

```diff
@@ -0,0 +1,159 @@
+package types
+
+import (
+	"bytes"
+	"encoding"
+	"fmt"
+	"time"
+)
+
+// Header abstracts all methods required to perform header sync.
+type H interface {
+	// New creates new instance of a header.
+	New() Header
+	// Hash returns hash of a header.
+	Hash() [32]byte
+	// Height returns the height of a header.
+	// Height() int64
+	// LastHeader returns the hash of last header before this header (aka. previous header hash).
+	LastHeader() [32]byte
+	// Time returns time when header was created.
+	// Time() time.Time
```

Comment:
it will be odd to just rename the two fields (Height and Time) to lower-case. Either we uniformly rename all fields (even then other structs like Block, Data, etc, have upper-case field names) or find derived names for header interface methods. e.g., `Time()` => `Timestamp()` and `Height()` => `Number()`. Basically, if you consider the Ethereum header, these alternatives are used. Is it possible to change the header interface this way?

Reply:
@tzdybal what is your opinion on this?

Reply:
Let's start with mixing getters with fields to match the header interface from https://github.com/celestiaorg/go-header. We plan to re-work the header in the near future anyways - then we will resolve this issue.